### ISSUE TITLE
[Vega 3] Canvas renderer doesn't fill arc marks correctly

### ISSUE THREAD

**Issue by clementprdhomme** [vega/vega#1158]
Hi everyone !

While building a pie chart with Vega 3, I noticed that when updating the fill colour of an arc, only part of the arc would get the new colour. After several attempts, my colleague noticed it doesn't happen with the SVG renderer.

Here is a screenshot of the issue:
<img width="839" alt="The biggest arc, originally yellow, is half filled with red" src="https://user-images.githubusercontent.com/6073968/36262993-ee167132-1260-11e8-99ee-f1605fe30ecf.png">

The biggest arc is originally yellow, but updated to red on hover. As you can see, only a part of it is turned red.

Here's the Vega Spec:
```json
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "width": 900,
  "height": 800,
  "data": [
    {
      "name": "table",
      "transform": [
        {
          "type": "window",
          "ops": ["row_number"],
          "as": ["rank"]
        },
        {
          "type": "formula",
          "as": "category",
          "expr": "datum.rank < 6 ? datum.x : 'Others'"
        },
        {
          "type": "aggregate",
          "groupby": ["category"],
          "ops": ["sum"],
          "fields": ["y"],
          "as": ["value"]
        },
        {
          "type": "pie",
          "field": "value",
          "startAngle": 0,
          "endAngle": 6.29
        }
      ],
      "values": [
        { "x": "Victor", "y": 84.38 },
        { "x": "Sergio", "y": 487.201 },
        { "x": "Rodrigo", "y": 493.4 },
        { "x": "Clément", "y": 857.64 },
        { "x": "Clara", "y": 874.4 },
        { "x": "Miguel", "y": 897.498 },
        { "x": "Tiago", "y": 976 },
        { "x": "Raúl", "y": 1987.49 },
        { "x": "Liuba", "y": 3946.57 },
        { "x": "Laura", "y": 9834.04 },
        { "x": "David", "y": 24328.42 },
        { "x": "Alicia", "y": 56979.49 }
      ]
    }
  ],
  "scales": [
    {
      "name": "c",
      "type": "ordinal",
      "range": "category",
      "domain": { "data": "table", "field": "category" }
    }
  ],
  "marks": [
    {
      "type": "arc",
      "from": { "data": "table" },
      "encode": {
        "enter": {
          "fill": { "scale": "c", "field": "category" },
          "x": { "signal": "width / 2" },
          "y": { "signal": "height / 2" }
        },
        "update": {
          "fill": { "scale": "c", "field": "category" },
          "opacity": { "value": 1 },
          "startAngle": { "field": "startAngle" },
          "endAngle": { "field": "endAngle" },
          "innerRadius": { "signal": "width > height ? height / 3 : width / 3" },
          "outerRadius": { "signal": "width > height ? height / 2 : width / 2" }
        },
        "hover": {
          "fill": { "value": "red" }
        }
      }
    }
  ]
}
```

**Please make sure to use the Canvas renderer to reproduce the issue.**

Any help with that?

Thank you!

**Comment by clementprdhomme** [comment]
Forgot to precise: I'm using Vega 3.0.10.

**Comment by jheer** [comment]
Thanks! You uncovered an arc bounds calculation error. This has been fixed and released as part of vega-scenegraph@2.3.1, and will be included in the upcoming Vega 3.1 release.
